comp_loinc/build.py

```diff
@@ -1,83 +1,173 @@
+):
+    """Build composed classes ontology.  Part 3/5 of the pipeline.
+
+    :param schema_file: Path to LinkML `.yaml` file that defines data model for "grouping classes" of LOINC terms, that 
+    is, classes that group sets of LOINC terms into specific categories.
+    :param composed_classes_data_file: Path to `.yaml` file which lists LOINC composed classes. These are lower-level,
+    more granular groupings of classes, and their are a greater number of them than the grouping classes in the
+    `schema_file`.
+    :param output: Path where output will be saved.
+
+    todo: this is just calling a linkml owl cli, should be written as code with python code
     """
     subprocess.call(["linkml-data2owl", "-s", schema_file, composed_classes_data_file, "-o", output])
 
+
 @app.command(name="merge")
 def merge_owl(
-        owl_directory: str,
-        output: str):
-    # TODO: Consider removing the files created from this point on each time this code executes e.g. any file with 'merge_*'
-    files = [f'{owl_directory}{x}' for x in os.listdir(owl_directory) if ".owl" in x]
+    owl_directory: Path = typer.Option(default=DEFAULTS['owl_directory'], resolve_path=True, exists=False),
+    output: Path = typer.Option(default=DEFAULTS['output.merge'], resolve_path=True, writable=True)
+):
+    """Merge all OWL ontology files into a single ontology. Part 4/5 of the pipeline.
+
+    :param owl_directory: Path to directory where unmerged `.owl` files are stored.
+    :param output: Path where output will be saved.
+
+    TODO: Consider removing the files created from this point each time this code executes e.g. any file with 'merge_*'
+    """
+    files = [f'{owl_directory}{x}' for x in os.listdir(owl_directory) if ".owl" in str(x)]
     subprocess.call(["./robot", "merge", "-i"] + " -i ".join(files).split() + ['-o', output])
 
+
 @app.command(name="reason")
 def reason_owl(
-        owl_directory: str,
-        merged_owl: str,
-        owl_reasoner: str,
-        output: str):
-    call_list = ["./robot", "reason", "-r", owl_reasoner, '-i', f"{owl_directory}{merged_owl}", '-o', f"{owl_directory}{output}"]
+    merged_owl: str = typer.Option(default=DEFAULTS['merged_owl']),
+    owl_reasoner: str = typer.Option(default=DEFAULTS['owl_reasoner']),
+    output: Path = typer.Option(default=DEFAULTS['output.reason'], resolve_path=True, writable=True)
+):
+    """Add computational reasoning to the merged ontology. Creates a new, reasoned ontology. Part 5/5 of the pipeline.
+
+    :param merged_owl: Name of the merged OWL file created from the `merge` command.
+    :param owl_reasoner: The name of the OWL reasoner to use.
+    :param output: Path where output will be saved."""
+    call_list = [
+        "./robot", "reason", "-r", owl_reasoner, '-i', f"{merged_owl}", '-o', f"{output}"]
     subprocess.call(call_list)
 
 
+# TODO: Add a comment about UX issue; how I needed to explicitly add all args
+@app.command(name="all")
+def run_all():
```

Comment:
I wanted this function to just make a call to each step without passing params, e.g. `build_part_ontology()`, but there is a bug or UX issue with `typer`, where if you call a function directly instead of over the command line, the default arguments do not populate.